webpush/encrypt.go

```diff
@@ -241,11 +292,42 @@ func newInfo(infoType string, context []byte) []byte {
 	info = append(info, []byte("Content-Encoding: ")...)
 	info = append(info, []byte(infoType)...)
 	info = append(info, 0)
-	info = append(info, []byte("P-256")...)
-	info = append(info, context...)
+	// For aes128gcm ctx is not needed.
+	if len(context) > 0 {
+		info = append(info, []byte("P-256")...)
+		info = append(info, context...)
+	}
 	return info
 }
 
+// Returns a keying material. See sections of
+// https://tools.ietf.org/html/draft-ietf-webpush-encryption-07#section-3.3
+func newKeyInfo(userAgentPublicKey, serverPublicKey []byte) []byte {
```

Comment:
chose subscriptionPublicKey to clarify it is retrieved from the PushSubscription